### PR TITLE
CMake build system.

### PR DESCRIPTION
**Pull request by vkalintiris**
##### Summary

Drop autotools and use CMake.

##### Test Plan

- CI jobs,
- Manual testing.

### CODE REVIEW

File: netdata-installer.sh
Comment (vkalintiris) [inline netdata-installer.sh:300]: `s/DEABLE_PLUGIN_FREEIPMI/DENABLE_PLUGIN_FREEIPMI/`

File: netdata-installer.sh
Comment (vkalintiris) [inline netdata-installer.sh:301]: ditto

File: .github/dockerfiles/Dockerfile.clang
Comment (tkatsoulas) [inline .github/dockerfiles/Dockerfile.clang:1]: I think this is the old docker images, why we don't drop this?

Comment (Ferroin) [reply .github/dockerfiles/Dockerfile.clang:1]: No, this is used as part of the checks workflow. It actually _should_ have been getting updated properly on every Debian release, but that’s obviously not happened.

**Comment by vkalintiris** [comment]
@Ferroin the installer tries to build local-listeners on FreeBSD which is not supported.

**Comment by vkalintiris** [comment]
> @vkalintiris,
> 
> Please, rebase your PR. I was testing your PR now, and I observed that it is possible to compile it using `netdata-installer.sh`, but it fails when I use `cmake` with the following error:
> 
> ```shell
> -- kinesis exporting: enabled
> -- pubsub exporting connector: disabled (requires grpc and googleapis)
> -- prometheus remote write exporting: disabled (requires protobuf and snappy libraries)
> -- mongodb exporting: enabled
> CMake Error at /usr/share/cmake-3.27/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
>   Could NOT find Protobuf (missing: Protobuf_LIBRARIES Protobuf_INCLUDE_DIR)
> Call Stack (most recent call first):
>   /usr/share/cmake-3.27/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
>   /usr/share/cmake-3.27/Modules/FindProtobuf.cmake:652 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
>   CMakeLists.txt:1221 (find_package)
> ```
> 
> best regards!

Can you copy/paste the exact command that you tried and failed?

**Comment by vkalintiris** [comment]
@Ferroin can you take a look at `Build / Test Source Build (alpine316, alpine:3.16, apk add -U bash , apk del json-c-dev , 3.16)`. It's not clear to me why it fails.

File: netdata-installer.sh
Comment (vkalintiris) [inline netdata-installer.sh:1178]: @Ferroin This check will return true on centos 7.

The cups plugin relies on API version >= 1.7. You can use `cups-config --api-version` to get the version.

File: netdata-installer.sh
Comment (vkalintiris) [inline netdata-installer.sh:1263]: @Ferroin 

`--install` was first added in 3.15:

https://cmake.org/cmake/help/v3.15/manual/cmake.1.html#cmdoption-cmake-install

Building from source will fail on Debian 10, because:

```
$ cmake --vesion
cmake version 3.13.4
```

We might be able to get away with using `ninja install` or `make install`. However, the proper fix is to require cmake >= `3.15`.

**Comment by vkalintiris** [comment]
@Ferroin There are still jobs running but I think it's safe to assume that only the static build fails now.

**Comment by thiagoftsm** [comment]
@vkalintiris ,

Please, rebase your PR.
I tested with success this https://github.com/netdata/netdata/pull/16548 PR on FreeBSD, but when I tried to test this it failed with [these errors](https://github.com/netdata/netdata/files/13592611/err.txt).

Best regards!

**Comment by thiagoftsm** [comment]
After last commits, it is possible to compile on Raspberry 4 and FreeBSD.

**Comment by thiagoftsm** [comment]
I retest after last commits on Slackware current, Raspberry 4 and FreeBSD. Everything is working as expected after last commits.

**Comment by vkalintiris** [comment]
@Ferroin Reverted:

- https://github.com/netdata/netdata/pull/15996/commits/4a915449d7a93d42177d7dd24042c161639cc17b
- https://github.com/netdata/netdata/pull/15996/commits/ff39cb8851a5fc962b530f0ac81d89a43c4f5d0f

They broke fluent-bit builds. Please, hold off until we merge this PR before modifying the way we are building external/bundled deps.

**Review by tkatsoulas — APPROVED** [review]

**Review by ilyam8 — APPROVED** [review]

**Pull request merged**
